Enhance frida ios dump, fix aapt and add gplaycli (#1455)
OWASP/owasp-mstg: Document/0x06b-Basic-Security-Testing.md

Successfully downloaded /var/mobile/Containers/Data/Application/72C7AAFB-1D75-4FBA-9D83-D8B4A2D44133/.com.apple.mobile_container_manager.metadata.plist to .com.apple.mobile_container_manager.metadata.plist
```

> You can also upload files with `file upload <local_file_path>` to the iOS device, but this implementation is not fully stable at the moment and might produce an error. If that's the case, file an issue in the [objection GitHub repo](https://github.com/sensepost/objection/issues "Objection Issues").

Reviewer: ```suggestion
You can also upload files to the iOS device with `file upload <local_file_path>`, but this implementation is not fully stable at the moment and might produce an error. If that's the case, file an issue in the [objection GitHub repo](https://github.com/sensepost/objection/issues "Objection Issues").
```

Reviewer: Isn't the unstable part always an issue? Don't we have a disclaimer somewhere at the start saying that tools can be broken and that they should open an issue / search github issues if that's the case? It also makes the guide very time specific, as it would have to be rewritten with every tool update and stuff like that.

I would remove the entire "but..." part.

Author: This is a very good point to address. This info was already there. And I'm afraid that we have this kind of sentences on more places in the guide.

I totally agree with your point, we should make it as timeless as possible. I'll check if we have a disclaimer, if not, we should definitely add one.

Author: I can't find it I'll add one proposal on a separate PR, it might need some discussion.

//
```

> In order to retrieve the unencrypted version, we can use tools such as [frida-ios-dump](https://github.com/AloneMonkey/frida-ios-dump "frida-ios-dump") or [Clutch](https://github.com/KJCracks/Clutch "Clutch"). Both will extract the unencrypted version from memory while the application is running on the device. The stability of both Clutch and Frida can vary depending on your iOS version and Jailbreak method, so it's useful to have multiple ways for extracting the binary. In general, all iOS versions lower than 12 should work with Clutch, while iOS 12+ should work with frida-ios-dump or a modified version of Clutch as discussed later.

Reviewer: ```suggestion
In order to retrieve the unencrypted version, we can use tools such as [frida-ios-dump](https://github.com/AloneMonkey/frida-ios-dump "frida-ios-dump") or [Clutch](https://github.com/KJCracks/Clutch "Clutch"). Both will extract the unencrypted version from memory while the application is running on the device. The stability of both Clutch and Frida can vary depending on your iOS version and Jailbreak method, so it's useful to have multiple ways of extracting the binary. In general, all iOS versions lower than 12 should work with Clutch, while iOS 12+ should work with frida-ios-dump or a modified version of Clutch as discussed later.
```

###### Using Frida-ios-dump

> [Frida-ios-dump](https://github.com/AloneMonkey/frida-ios-dump "Frida-ios-dump") is a Python script that helps you retrieving the decrypted version of an iOS app to your host machine. It supports both Python 2 and Python 3 and requires Frida running on your iOS device (jailbroken or not). It is basically injecting a JavaScript script (`dump.js`) that uses Frida's [Memory API](https://www.frida.re/docs/javascript-api/#memory "Frida Memory API") to dump the memory of the running app onto a file (as it is already decrypted by iOS).

Reviewer: ```suggestion
[Frida-ios-dump](https://github.com/AloneMonkey/frida-ios-dump "Frida-ios-dump") is a Python script that helps you retrieve the decrypted version of an iOS app from an iOS device. It supports both Python 2 and Python 3 and requires Frida running on your iOS device (jailbroken or not). This tool uses Frida's [Memory API](https://www.frida.re/docs/javascript-api/#memory "Frida Memory API") to dump the memory of the running app and recreate an IPA file. Because the code is extracted from memory, it is automatically decrypted.
```

###### Using Frida-ios-dump

> [Frida-ios-dump](https://github.com/AloneMonkey/frida-ios-dump "Frida-ios-dump") requires Frida server running on your jailbroken device. It is basically using Frida script (`dump.js`) to dump the decrypted binary from memory onto a file.

Reviewer: Not for the packaged version (objection patchipa style)

Author: While that's true and it would surely work with the patched IPA on a non-jailbroken device. It would only make sense if you have an encrypted IPA that you would patch, install and decrypt. Can you think of a scenario where this would be the case? We could add it as a note.

Author: please recheck these lines, I've rewritten them.

> > Frida-ios-dump might not be compatible with the latest version of Frida. If that's the case, you might have to install an older version of Frida server on your device.

Reviewer: Same not about tools being potentially broken. I would say, unless there is very specific <version or  >version info, we shouldn't mention it.

Author: Agree. We shouldn't maintain this kind of things. It should be of common sense to take a look in the repo for (known) issues and for creating them if something fails.

> > In addition, note that we'll be using the master branch of frida-ios-dump, which runs on Python 2. If you have any issues with it you might try checking out the 3.x branch and run frida-ios-dump again using Python 3.

Reviewer: ```suggestion
> In addition, note that the examples below are using the master branch of frida-ios-dump, which runs on Python 2. If you have any issues with it you might try checking out the 3.x branch and run frida-ios-dump again using Python 3.
```
